packages/expo-updates/ios/EXUpdates/EXUpdatesAppController.m

```diff
@@ -251,36 +250,68 @@ - (void)appLoaderTask:(EXUpdatesAppLoaderTask *)appLoaderTask didFinishBackgroun
   }
 }
 
-# pragma mark - internal
+# pragma mark - EXUpdatesAppController+Internal
 
-- (EXUpdatesConfig *)_loadConfigFromExpoPlist
+- (BOOL)initializeUpdatesDirectoryWithError:(NSError ** _Nullable)error
 {
-  NSString *configPath = [[NSBundle mainBundle] pathForResource:EXUpdatesConfigPlistName ofType:@"plist"];
-  if (!configPath) {
-    @throw [NSException exceptionWithName:NSInternalInconsistencyException
-                                   reason:@"Cannot load configuration from Expo.plist. Please ensure you've followed the setup and installation instructions for expo-updates to create Expo.plist and add it to your Xcode project."
-                                 userInfo:@{}];
+  NSError *err;
+  _updatesDirectory = [EXUpdatesUtils initializeUpdatesDirectoryWithError:&err];
+  if (err && error) {
+    *error = err;
   }
+  return _updatesDirectory != nil;
+}
 
-  return [EXUpdatesConfig configWithDictionary:[NSDictionary dictionaryWithContentsOfFile:configPath]];
+- (BOOL)initializeUpdatesDatabaseWithError:(NSError ** _Nullable)error
+{
+  __block BOOL dbSuccess;
+  __block NSError *dbError;
+  dispatch_semaphore_t dbSemaphore = dispatch_semaphore_create(0);
+  dispatch_async(_database.databaseQueue, ^{
+    dbSuccess = [self->_database openDatabaseInDirectory:self->_updatesDirectory withError:&dbError];
+    dispatch_semaphore_signal(dbSemaphore);
+  });
+
+  dispatch_semaphore_wait(dbSemaphore, DISPATCH_TIME_FOREVER);
+  if (dbError && error) {
+    *error = dbError;
+  }
```

Comment:
```suggestion
  if (dbError) {
    *error = dbError;
  }
```
A non-null error is passed to `initializeUpdatesDatabaseWithError` at every call site, is this some quirk of OBJ-C that you are catering to?

Reply:
Callers may pass in nil

Reply:
oh true 🙃